src/packet_sources/macos.rs

```diff
@@ -0,0 +1,229 @@
+    ip_rx: pipe::Receiver,
+    ip_tx: pipe::Sender,
+    net_rx: pipe::Receiver,
+    net_tx: pipe::Sender,
+    filter_rx: pipe::Receiver,
+    filter_tx: pipe::Sender,
+    ip_path: PathBuf,
+    net_path: PathBuf,
+    filter_path: PathBuf,
+}
+
+impl PipeServer {
+    pub async fn new(ip_pipe: &str, net_pipe: &str, filter_pipe: &str) -> Result<Self> {
+        let home_dir = match home_dir() {
+            Some(ip_path) => ip_path,
+            None => Err(anyhow!("Failed to get home directory"))?,
+        };
+
+        let ip_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &ip_pipe));
+        let net_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &net_pipe));
+        let filter_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &filter_pipe));
+
+        let (ip_rx, ip_tx) = Self::create_pipe(&ip_path)?;
+        let (net_rx, net_tx) = Self::create_pipe(&net_path)?;
+        let (filter_rx, filter_tx) = Self::create_pipe(&filter_path)?;
+
+        Ok(PipeServer {
+            ip_rx,
+            ip_tx,
+            net_rx,
+            net_tx,
+            filter_rx,
+            filter_tx,
+            ip_path,
+            net_path,
+            filter_path,
+        })
+    }
+
+    fn create_pipe(path: &PathBuf) -> Result<(pipe::Receiver, pipe::Sender)> {
+        if !path.exists() {
+            match mkfifo(path, Mode::S_IRWXU) {
+                Ok(_) => println!("created {:?}", path),
+                Err(e) => Err(anyhow!("Failed to create fifo: {:?}", e))?,
+            }
+        }
+        let rx = match pipe::OpenOptions::new()
+            .unchecked(true)
+            .open_receiver(&path)
+        {
+            Ok(rx) => rx,
+            Err(e) => Err(anyhow!("Failed to open fifo receiver: {:?}", e))?,
+        };
+        let tx = match pipe::OpenOptions::new().unchecked(true).open_sender(&path) {
+            Ok(tx) => tx,
+            Err(e) => Err(anyhow!("Failed to open fifo receiver: {:?}", e))?,
+        };
+
+        Ok((rx, tx))
+    }
+}
+
+pub struct MacosConf;
+
+#[async_trait]
+impl PacketSourceConf for MacosConf {
+    type Task = MacosTask;
+    type Data = UnboundedSender<ipc::FromProxy>;
+
+    fn name(&self) -> &'static str {
+        "Macos proxy"
+    }
+
+    async fn build(
+        self,
+        net_tx: Sender<NetworkEvent>,
+        net_rx: Receiver<NetworkCommand>,
+        sd_watcher: broadcast::Receiver<()>,
+    ) -> Result<(MacosTask, Self::Data)> {
+        let executable_path = "/Applications/MitmproxyAppleTunnel.app/";
+
+        let ipc_server = match PipeServer::new("ip", "net", "filter").await {
+            Ok(server) => server,
+            Err(e) => Err(anyhow!("Failed to create pipe server: {:?}", e))?,
+        };
+
+        let result = Command::new("open")
+            .arg("-a")
+            .arg(executable_path)
+            .arg("--args")
+            .arg(&ipc_server.ip_path)
+            .arg(&ipc_server.net_path)
```

Comment:
we can Avoid to send net_path, I will remove it in next swift small refactor (when I will fix protobuf on swift app